docs/uml.py

```diff
@@ -0,0 +1,467 @@
+
+    @staticmethod
+    def _remove_last_package(namespace: str) -> str:
+        return f'{".".join(namespace.split(".")[0:-2])}.{namespace.split(".")[-1]}'
+
+
+class PlantUMLNetwork(_UMLNetworkABC):
+    """
+    UML-Network-Parser inheriting from `_UMLNetworkABC`. This subclass parses the network (i.e. itself) into plantuml
+    output.
+    """
+
+    def _node_to_str(self, node: str, detailed: bool = True, root_node: Optional[str] = None) -> str:
+        """
+        Parse the class with their fields into a string for plantuml output.
+        The class will contain a link to the respective class in the documentation.
+        """
+        cls_str = (
+            f'class "[[{LINK_DOMAIN}/api/bo4e.{node.split(".")[1]}.html#{node}'
+            f' {node.split(".")[-1]}]]\\n<size:10>{".".join(node.split(".")[0:-1])}" as {node.split(".")[-1]}'
+        )
+        if detailed:
+            cls_str += " {\n"
+            for field in self.nodes[node]["fields"].values():
+                type_str = _UMLNetworkABC.model_field_str(field)
+                if field.required:
+                    cls_str += f"\t{field.alias} : {type_str}\n"
+                else:
+                    cls_str += f"\t{field.alias} : {type_str} = {field.default}\n"
+            cls_str += "}"
+
+        return cls_str
+
+    # pylint: disable=too-many-arguments, too-many-locals
+    def _edge_to_str(self, node1: str, node2: str, index: int, detailed: bool, root_node: Optional[str] = None) -> str:
+        """
+        Parse the connection into a string for plantuml output.
+        """
+        vert_extension = "{node1} -[norank]-|> {node2} #line:gray"
+        horz_extension_lr = "{node1} --|> {node2}"
+        horz_extension_rl = "{node2} <|-- {node1}"
+        vert_association = "{node1} {card1} -[norank]-* {card2} {node2} #line:gray;text:gray : {field}"
+        horz_association_lr = "{node1} {card1} --* {card2} {node2} : {field}"
+        horz_association_rl = "{node2} {card2} *-- {card1} {node1} : {field}"
+
+        # The only purpose of this is to use less code below by using the get-method with default values
+        association: Dict[Optional[str], str] = {
+            node1: horz_association_lr,
+            node2: horz_association_rl,
+        }
+        extension: Dict[Optional[str], str] = {
+            node1: horz_extension_lr,
+            node2: horz_extension_rl,
+        }
+
+        node1_str = _UMLNetworkABC._remove_last_package(node1)
+        node2_str = _UMLNetworkABC._remove_last_package(node2)
+        if root_node == node1:
+            node1_str = "." + node1.split(".")[-1]
+        elif root_node == node2:
+            node2_str = "." + node2.split(".")[-1]
+
+        if self[node1][node2][index]["type"] == "extension":
+            return extension.get(root_node, vert_extension).format(node1=node1_str, node2=node2_str)
+        if self[node1][node2][index]["type"] == "association":
+            card1 = ""
+            card2 = ""
```

Comment:
```suggestion
            # "card" is short for cardinality
            card1 = ""
            card2 = ""
```
für die unwissenden, ich hab das erst im debugger erkannt.

Reply:
Sorry, wenn man den Code schreibt, hält man sowas gerne mal für offensichtlich, was die Abkürzungen bedeuten sollen ^^'